pkg/cloud/azure/services/certificates/certificates.go

```diff
@@ -40,24 +42,25 @@ import (
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 	"k8s.io/kubernetes/cmd/kubeadm/app/util/pubkeypin"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure"
 )
 
 // Get should implement returning certs and kubeconfigs.
-func (s *Service) Get(ctx context.Context) (interface{}, error) {
+func (s *Service) Get(ctx context.Context, spec azure.Spec) (interface{}, error) {
 	return nil, errors.New("Not implemented")
 }
 
 // Delete cleans up and generated certificates, could be useful for renewal.
-func (s *Service) Delete(ctx context.Context) error {
+func (s *Service) Delete(ctx context.Context, spec azure.Spec) error {
 	return nil
 }
 
 // CreateOrUpdate Helper function so this can be unittested.
-func (s *Service) CreateOrUpdate(ctx context.Context) error {
+func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
+	klog.V(2).Infof("generating certificates")
 	clusterName := s.scope.Cluster.Name
 	tmpDirName := "/tmp/cluster-api/" + clusterName
-	dnsName := fmt.Sprintf("%s-api", clusterName)
+	dnsName := fmt.Sprintf("%s.%s.cloudapp.azure.com", strings.ToLower(s.scope.ClusterConfig.PublicIPName), strings.ToLower(s.scope.ClusterConfig.Location))
```

Comment:
This needs to be support non-Azure DNS names e.g., when we move to the point where DNS is not provider by Azure DNS.

Reply:
I have moved it to defaults, since in future we want to move all defaults to be configurable, this will be taken care of.